x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Kql.java

```diff
@@ -62,12 +79,22 @@ public String getWriteableName() {
 
     @Override
     public Expression replaceChildren(List<Expression> newChildren) {
-        return new Kql(source(), newChildren.get(0));
+        return new Kql(source(), newChildren.get(0), queryBuilder());
     }
 
     @Override
     protected NodeInfo<? extends Expression> info() {
-        return NodeInfo.create(this, Kql::new, query());
+        return NodeInfo.create(this, Kql::new, query(), queryBuilder());
     }
 
+    @SuppressWarnings("rawtypes")
+    @Override
+    protected ExpressionTranslator translator() {
+        return new EsqlExpressionTranslators.KqlFunctionTranslator();
```

Comment:
I guess we could move the translators from `EsqlExpressionTranslators` but for now I just leave them as they are - I wasn't sure we need to move them. happy to hear otherwise.

Reply:
My preference is to move them, but not in this PR - as a follow up.